internal/packetfabric/ports.go

```diff
@@ -333,6 +333,15 @@ func (c *PFClient) UpdatePort(autoNeg bool, portCID, description string) (*Inter
 	if err != nil {
 		return nil, err
 	}
+	updateOk := make(chan bool)
+	defer close(updateOk)
+	fn := func() (*ServiceState, error) {
+		return c.GetPortStatus(expectedResp.PortCircuitID)
+	}
+	go c.CheckServiceStatus(updateOk, fn)
+	if !<-updateOk {
+		return nil, err
+	}
```

Comment:
can we make it one function, i see it is the same even below

Reply:
Good suggestion. Because I want to release today though. I will keep this as is but we can optimize in the future. I will create a issue on GitHub https://github.com/PacketFabric/terraform-provider-packetfabric/issues/367. I think we could do the same other places where we check status of services creation/update/delete.